src/meta.rs

```diff
@@ -0,0 +1,274 @@
+use std::collections::HashMap;
+
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct Meta {
+    fields: Fields,
+}
+
+impl Meta {
+    /// Get a reference to the named field position metadata.
+    #[must_use]
+    pub fn fields(&self) -> &Fields {
+        &self.fields
+    }
+
+    /// Get a mutable reference to the named field position metadata.
+    pub fn fields_mut(&mut self) -> &mut Fields {
+        &mut self.fields
+    }
+}
+
+/// The metadata and inner [Fields] of a field.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct Field {
+    meta: String,
+    fields: Option<Fields>,
+}
+
+impl Field {
+    /// Create a new empty field metadata.
+    #[must_use]
+    pub const fn empty() -> Self {
+        Self {
+            meta: String::new(),
+            fields: None,
+        }
+    }
+
+    /// Create a new field metadata.
+    pub fn new(meta: impl Into<String>, fields: Option<Fields>) -> Self {
+        Self {
+            meta: meta.into(),
+            fields,
+        }
+    }
+
+    /// Get the metadata of this field.
+    #[must_use]
+    pub fn meta(&self) -> &str {
```

Comment:
Since I'm still thinking about generalising this feature in the future, I think that meta should not be string-typed but that there should be a meta struct with private members (for now just a string) and a method to set a doc comment (since that's the metadata we support right now). A later extension could then add further private fields and public methods to this struct